src/components/Stack/Stack.stories.tsx

```diff
@@ -0,0 +1,158 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import React, { ReactElement, ReactNode } from 'react';
+import { Meta, StoryFn } from '@storybook/react';
+import { STACK_PROPS, Stack as StackComponent } from './Stack';
+import { SPACING_VALUES } from '@utilities/dimensions';
+import { BOX_BG_ALIAS_TOKENS_CLASSES, BOX_TEXT_ALIAS_TOKENS_CLASSES } from '@utilities/tokens';
+import { Box } from '@components/Box';
+import { Divider } from '@components/Divider';
+
+const STACK_DIRECTIONS = ['row', 'column', 'row-reverse', 'column-reverse'];
+
+const STACK_JUSTIFY_TYPES = ['start', 'end', 'center', 'between', 'around', 'evenly'];
+
+const STACK_ALIGN_ITEMS_TYPES = ['start', 'end', 'center', 'baseline', 'stretch'];
+
+const spacingSelect = {
+    options: Object.values(SPACING_VALUES),
+    control: { type: 'select' },
+};
+
+export default {
+    title: 'Layout/Stack',
+    component: StackComponent,
+    tags: ['autodocs'],
+    argTypes: {
+        as: {
+            options: ['div', 'span'],
+            control: {
+                type: 'select',
+            },
+        },
+        bg: {
+            options: BOX_BG_ALIAS_TOKENS_CLASSES,
+            control: {
+                type: 'select',
+            },
+        },
+        color: {
+            options: BOX_TEXT_ALIAS_TOKENS_CLASSES,
+            control: {
+                type: 'select',
+            },
+        },
+        margin: spacingSelect,
+        padding: spacingSelect,
+        direction: {
+            options: STACK_DIRECTIONS,
+            control: { type: 'select' },
+        },
+        spacing: {
+            options: SPACING_VALUES,
+            control: { type: 'select' },
+        },
+        justify: {
+            options: STACK_JUSTIFY_TYPES,
+            control: { type: 'select' },
+        },
+        alignItems: {
+            options: STACK_ALIGN_ITEMS_TYPES,
+            control: { type: 'select' },
+        },
+        children: { table: { disable: true } },
+    },
+    args: {
+        as: 'div',
+        bg: '',
+        color: '',
+        margin: 0,
+        padding: 12,
+        direction: 'column',
+        justify: 'start',
+        alignItems: 'start',
+        spacing: 8,
+        divider: <Divider height={'0px'} />,
+        'data-test-id': 'custom-test-id',
+    },
+} as Meta<STACK_PROPS>;
+
+const Code = ({ children }: { children: ReactNode }): ReactElement => (
+    <code className="tw-bg-box-neutral tw-rounded tw-px-2 tw-text-box-neutral-inverse tw-text-s">{children}</code>
+);
+
+export const Stack: StoryFn<STACK_PROPS> = (args: STACK_PROPS) => {
+    const isVertical = args.direction === 'row' || args.direction === 'row-reverse';
+
+    return (
+        <StackComponent {...args} divider={<Divider vertical={isVertical} />}>
```

Comment:
The Divider is enforced while it's an optional property. I would create different stories:
- Stack (No Divider)
- Stack with horizontal Divider
- Stack with vertical Divider